### ISSUE TITLE
Implementing 'with recursive'

### ISSUE THREAD

**Issue by hesselink** [tomjaguarpaw/haskell-opaleye#55]
Have you thought about implementing [with recursive](http://www.postgresql.org/docs/9.3/static/queries-with.html)? I'd like to perhaps give it a go, and was thinking it would look something like:

``` haskell
recursive :: Query a -> QueryArr a a -> Query a
```

This would then generate something like:

``` sql
with recursive temp (cols) (
  <query1>
  union all
  <query2 with references to the input bound to 'temp'>
  )
select cols from temp
```

Do you think that makes sense? Do you have any pointers for the implementation?

**Comment by hesselink** [comment]
I've taken a stab at this but could use some help in #58.

**Comment by tomjaguarpaw** [comment]
It's a good idea.  At one point I wondered whether it would be `loop :: QueryArr (a, d) (b, d) -> QueryArr a b`, but I think it would also require some typeclass constraint on `d` at least.

**Comment by dwarfmaster** [comment]
Since this has been solved in 6cc38b4, maybe this could be closed ? This issue was the first result I found when searching how to do with recursive with Opaleye and might lead people to believe it hasn't been implemented yet.

**Comment by tomjaguarpaw** [comment]
Yes, good idea. This has been in Opaleye since at least 0.9.3.0: https://hackage.haskell.org/package/opaleye-0.9.3.0/docs/Opaleye-With.html
